Feat/#56 DB 데이터 모델 재정의

ggoldJeongg opened the pull request:
## 🔘Part

- [x] BE

  <br/>

## 🔎 작업 내용
- 기존 코드가 잘못된 계층 분리로 로직이 일부만 수정되어도 데이터 조회가 되지않았습니다.
- 엔티티->DTO -> Repository -> 구현체 순서로 작업했습니다.
- 백엔드 네이밍 룰과 레포지토리가 잘 일치하는지, 인터페이스 정의가 괜찮은지 봐주시면 감사하겠습니다.


  <br/>

## 참고사항
- 기존 코드를 다른 브랜치로 가져왔기때문에 파일이 많습니다. 모든 코드를 보실 필요 없습니다.
- 0b566dbe6a1c0561c5de6295931d740ef59aca79 9f0da1dc9f17f9171ddf41dc8ab1929124b31fc8 9241189879fbe9ab2623412c0dcd4f337208c3ff 이 3가지만 봐주시면 됩니다.

<br/>

## 🔧 앞으로의 과제
- UI 컴포넌트 내부 상태관리 usecase 분리 
- API 호출 로직 구현
- UI 컴포넌트 구현

  <br/>

## ➕ 이슈 링크

- [소톡소톡 #56]

<br/>

ggoldJeongg committed:
Feat/#56 엔티티 타입 정의

ggoldJeongg committed:
Feat/#56 프로필 페이지 DTO 정의

ggoldJeongg committed:
Refactor/food, topic DTO 배열로 변경경

ggoldJeongg committed:
Add/#5 기존 5번 이슈 파일 추가가

ggoldJeongg committed:
Merge branch 'dev' of https://github.com/FRONT-END-BOOTCAMP-PLUS-3/obama into Feat/56-profile-page

ggoldJeongg committed:
Rename/#56 프로필 조회 페이지 entities 프로필 하위폴더로 이동

ggoldJeongg committed:
Refactor/#63 기존 프로필 SbRepository 코드 리팩토링

- 현재 코드에서는 유저 정보, 카테고리, 입력값, SNS 정보, 공개 여부 정보 조회가 하나의 메서드에서 수행되고 있어 책임이 과도하게 집중
- 단일 책임 원칙에 위배되어 각자 책임 단위로 repository 분리
- 분리된 repository별로 DI 생성성

ggoldJeongg committed:
Fix/코드리뷰 반영

- 기존 생성한 Category 관련 레포지토리 삭제 (category API를 재사용)
- 프로필 DTO tripStyle -> trip으로 테이블을 따라가도록 변경

getsoss commented on src/app/user/profile/components/PrivacyToggle.styled.tsx:
StyeldComponent에 쓰인 px 단위들 rem 으로 바꿔 주시면 좋을 것 같습니다 😃  수고 많으셨습니다!

getsoss commented on src/domain/repositories/profile/ISbUserinputRepository.ts:
파일명이 ISbUserInputRepository.ts 입니다! IUserInputRepository.ts 로 수정해주시면 감사하겠습니다 👍

ggoldJeongg committed:
Fix/코드리뷰 반영

- ISBUserinputRepository -> IUserinputRepository 파일명 수정
- styled-components 단위값 px -> rem으로 변경경

JunSeGue commented on src/app/api/profile/route.ts:
```diff
@@ -0,0 +1,46 @@
+
+  try {
+    // Axios를 사용하여 병렬 요청 처리
+    const [profileDetails, aboutMeData, privacySettings] = await Promise.all([
+      axios.get(`https://your-backend-api.com/profile/${userId}`).then((res) => res.data),
+      axios.get(`https://your-backend-api.com/about-me/${userId}`).then((res) => res.data),
+      axios.get(`https://your-backend-api.com/privacy-settings/${userId}`).then((res) => res.data),
+    ]);
```
adeptor에서 axios를 부를 필요가 없습니다. 사용되면 clean아키텍처 위반이라 ui에서 호출하게 해주세요

ggoldJeongg replied:
반영 완료

JunSeGue committed:
Delete src/app/page.tsx

JunSeGue commented on src/app/user/profile/components/AboutMe.tsx:
```diff
@@ -0,0 +1,37 @@
+        Object.entries(aboutMeData)
+          .filter(([_, values]) => values.length > 0 && values[0] !== "") // 값 필터링
+          .map(([label, values]) => (
+            <Item key={label}>
```
이곳에 쓰이는 label은 무엇인지 말씀해주세요

ggoldJeongg replied:
기존 리팩토링 반영 안된 코드입니다. 관련 리팩토링 안된 부분 삭제후 push 하겠습니다
label: category, value:userinput 값 이었습니다.
AboutMe 부분은 재연님 category 로직을 재사용할 예정입니다!

ggoldJeongg replied:
[반영 완료] 기존 리팩토링 반영 안된 코드입니다.
label: category, value:userinput 값 이었습니다.
AboutMe 부분은 재연님 category 로직을 재사용할 예정입니다!

JunSeGue commented on src/app/user/profile/components/PrivacyToggle.styled.tsx:
1. 폴더구조 요청! => src/components/porfile/"파일명"
2. 파일명 변경 요청!  => PrivacyToggle.Styled.tsx 
컨벤션 지켜주세요

ggoldJeongg replied:
반영 완료

JunSeGue commented on src/app/user/profile/components/ProfileDetails.tsx:
```diff
@@ -0,0 +1,100 @@
+    hobbies: true,
+    interests: false,
+  });
+
```
ui에서도 snake_case가 쓰인다는거면 dto도 snake_case인걸로 알고있는데 dto는 무조건 카멜입니다.

ggoldJeongg replied:
[반영 완료] ProfileHeader 컴포넌트로 로직을 옮길 예정이라서 해당 로직 삭제했습니다
해당 내용 반영해서 추후 개발 진행하겠습니다

JunSeGue commented on src/app/user/profile/components/ProfileDetails.tsx:
```diff
@@ -0,0 +1,100 @@
+      const data = await getProfileDetailsUsecase.execute(userId);
+      setProfileDetails(data);
+    };
+
```
adeptor가 ui에로직에 있는데 수정바랍니다.

ggoldJeongg replied:
[반영 완료] profileheader, aboutme 컴포넌트와 데이터 조회가 중복되어 삭제했습니다.

JunSeGue commented on src/app/user/profile/page.styled.tsx:
page.styled.tsx가 아닌 ProfilePage.tsx로 변경해주세요

ggoldJeongg replied:
반영 완료

JunSeGue commented on src/app/user/profile/page.tsx:
```diff
@@ -0,0 +1,52 @@
+      const repository = new SbProfileRepository();
+      const data = await repository.getProfile("1d1867cd-526c-4de5-97e4-4a0c8f386f78"); //테스트 사용자 ID 
+      setProfile(data);
+
```
여기도 마찬가지로 adeptor을 로직이 들어가있습니다.

ggoldJeongg replied:
반영 완료

JunSeGue commented on src/app/user/profile/page.tsx:
```diff
@@ -0,0 +1,52 @@
+    fetchProfile();
+  }, []);
+
+  if (!profile) return <div>사용자 정보를 찾을 수 없습니다.</div>;
```
if 시 사용자 정보를 찾을 수 없는게 아닌 craete페이지로 이동할수 있는데 페이지나 버튼 등이 있으면 좋을 것 같습니다.

ggoldJeongg replied:
해당 내용은 전체적인 코드 리팩토링을 통해서 조회 로직이 동작하면 새로운 feature로 추가하겠습니다

JunSeGue commented on src/application/profile/dtos/AboutMeDTO.ts:
src/application/usecases/profile에 넣으시면됩니다.

ggoldJeongg replied:
반영 완료

JunSeGue commented on src/application/profile/usecases/GetAboutMeUsecase.ts:
클린 아키텍처 구조 위반 repository 가 usecase랑 분리가 되어야한다.
server.ts에서 createClient를 수정해놔서 사용하지 않는 코드로 보입니다. 사용을 안하시면 삭제해주세요

JunSeGue commented on src/application/profile/usecases/GetPrivacySettingsUsecase.ts:
```diff
@@ -0,0 +1,21 @@
+  // 개인정보 설정 가져오기
+  async getPrivacySettings(userId: string): Promise<PrivacySettingDTO[]> {
+    return await this.privacySettingRepository.getPrivacySettings(userId);
+  }
```
getPrivacySettings하고 
updatePrivacySetting하고 서로 다른 역활을 하고 있는데
usecase를 나눠주시면 더욱 편한 로직이 될것입니다.

ggoldJeongg replied:
반영 완료

JunSeGue commented on src/application/profile/usecases/GetProfileDetailsUsecase.ts:
```diff
@@ -0,0 +1,36 @@
+      .from("user")
+      .select("name, birth_date, phone, email")
+      .eq("user_id", userId)
+      .single();
```
usecase에서 db에 접근해서 데이터를 가져오지 않습니다.
클린 아키텍처 구조로 변경!!!

ggoldJeongg replied:
[반영 완료] profileheader, aboutme와 중복 로직으로 삭제했습니다.

JunSeGue commented on src/domain/entities/profile/PrivacySetting.ts:
DB형태 변경으로 인한entity수정

ggoldJeongg replied:
수정 완료

JunSeGue commented on src/infrastructure/repositories/profile/SbPrivacySettingRepository.ts:
db형태 수정으로 인한 수정사항으로 알고있음

ggoldJeongg replied:
이 부분은 66번 이슈 해결하면서 같이 리팩토링 진행하도록하겠습니다.

JunSeGue reviewed (changes requested):
1. 클린아키텍처 반영 안된 코드들도 같이 push가 되니 명심해주세요
2. 폴더 구조에 맞춰주세요

ggoldJeongg committed:
Rename/#56 코드리뷰 반영

- user>profile 컴포넌트에서 components>profile 컴포넌트로 폴더구조 이동
- 스타일시트 컨벤션 준수 (Styled.tsx)

ggoldJeongg committed:
Remove/중복 로직 파일 삭제

- 일부 유저 정보 조회 로직이 profileheader, aboutme 컴포넌트와 중복되어 삭제

ggoldJeongg committed:
Fix/#65 코드리뷰 반영

- PrivacySetting 엔티티 DB 변경으로 스네이크케이스로 변경 (스네이크케이스 ->카멜메이스로 변환 할것)
- 프로필 페이지 api 호출 로직 변경
- PrivacySettingUsecase 분리 (get,update)
- usecase 폴더 구조 변경

ggoldJeongg committed:
Merge branch 'dev' of https://github.com/FRONT-END-BOOTCAMP-PLUS-3/obama into Feat/56-profile-page

JunSeGue commented on src/domain/entities/profile/PrivacySetting.ts:
isPublic이 테이블이 있는데 써야하는 테이블을 정의한 이유가 뭐죠?

ggoldJeongg replied:
[반영 완료] 기존 리팩토링 안된 코드로 불필요하다고 판단되어 삭제했습니다.

JunSeGue commented on src/app/api/profile/route.ts:
의존성 주입을 개별 지역에서 선언하게해주세요

ggoldJeongg committed:
Refactor/#65 엔티티 스네이크 케이스 변경, 불필요한 엔티티 삭제

- 엔티티 카멜케이스로 변경
- 블필요한 isPublic 테이블로 대체 가능한 PrivacySettings 엔티티 삭제제

ggoldJeongg committed:
Refactor/#65 프로필 개인정보 설정 및 조회 api, usecase 수정

- `GET /api/profile` 요청 시 userId 기반으로 프로필 조회 (userid 추가 수정 필요요)
- usecase 일부 경로 및 폴더명 수정

ggoldJeongg committed:
Refactor/#65 프로필 DTO에 기존 엔티티 재활용용

ggoldJeongg committed:
Refactor/#65 PrivacySettingRepository 삭제 -> Ispublic 테이블로 변경

JunSeGue committed:
Update PrivacySettingsUsecase.ts

JunSeGue committed:
Update UpdatePrivacySettingUsecase.ts

ggoldJeongg committed:
Merge branch 'dev' into Feat/56-profile-page
